File: 0724-find-pivot-index/0724-find-pivot-index.py
# sol1- GPT4

class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        total_sum = sum(nums)  # Total sum of the array
        left_sum = 0  # Sum of elements to the left of the current index

        # Iterate through the array to find the pivot index
        for idx, num in enumerate(nums):
            # Check if the left sum equals the right sum (total_sum - left_sum - num)
            if left_sum == total_sum - left_sum - num:
                return idx
            left_sum += num

        return -1  # Return -1 if no pivot index found

File: 0724-find-pivot-index/0724-find-pivot-index.c
// sol1 - GPt4

int pivotIndex(int* nums, int numsSize) {
    int total_sum = 0;  // Total sum of the array
    for (int i = 0; i < numsSize; i++) {
        total_sum += nums[i];
    }

    int left_sum = 0;  // Sum of elements to the left of the current index

    // Iterate through the array to find the pivot index
    for (int idx = 0; idx < numsSize; idx++) {
        // Check if the left sum equals the right sum (total_sum - left_sum - nums[idx])
        if (left_sum == total_sum - left_sum - nums[idx]) {
            return idx;
        }
        left_sum += nums[idx];
    }

    return -1;  // Return -1 if no pivot index found
}
